File: src/main/java/com/example/demo/mapper/ReportJobMapper.java
package com.example.demo.mapper;

import com.example.demo.model.ReportJob;
import org.apache.ibatis.annotations.Mapper;

import java.util.List;
import java.util.Map;

@Mapper
public interface ReportJobMapper {
    int insert(ReportJob job);

    ReportJob selectById(Long id);

    int updateStatus(Map<String, Object> params);

    List<ReportJob> selectPending();
}

File: src/main/kotlin/com/example/demo/mapper/ReportJobMapper.kt
package com.example.demo.mapper

import com.example.demo.model.ReportJob
import org.apache.ibatis.annotations.Mapper

@Mapper
interface ReportJobMapper {
    fun insert(job: ReportJob): Int
    fun selectById(id: Long): ReportJob?
    fun updateStatus(params: Map<String, Any?>): Int
    fun selectPending(): List<ReportJob>
}

